packages/experimental/kai/src/app/pages/Sidebar.tsx

```diff
@@ -3,25 +3,42 @@
 //
 
 import { PlusCircle, Gear, Robot, Trash, WifiHigh, WifiSlash } from 'phosphor-react';
-import React, { useMemo } from 'react';
-import { useNavigate } from 'react-router-dom';
+import React, { useMemo, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
 
 import { ConnectionState } from '@dxos/protocols/proto/dxos/client/services';
 import { useClient, useNetworkStatus } from '@dxos/react-client';
 import { getSize, mx } from '@dxos/react-components';
+import { useTogglePanelSidebar } from '@dxos/react-ui';
 
 import { Button } from '../../components';
 import { MemberList, SpaceList } from '../../containers';
-import { useSpace } from '../../hooks';
+import { AppView, useSpace } from '../../hooks';
 import { Generator } from '../../proto';
 
 export const Sidebar = () => {
+  const toggleSidebar = useTogglePanelSidebar();
   const navigate = useNavigate();
   const client = useClient();
   const { space } = useSpace();
   const { state: connectionState } = useNetworkStatus();
   const generator = useMemo(() => (space ? new Generator(space.experimental.db) : undefined), [space]);
 
+  const { view } = useParams();
+  const [prevView, setPrevView] = useState(view);
+  const [prevSpace, setPrevSpace] = useState(space);
+
+  // TODO(wittjosiah): Find a better way to do this.
```

Comment:
What's the concept here?

Reply:
This closes the sidebar when a new space is selected. This is needed for mobile to feel not broken because when the sidebar is open it covers most of the page.

I think ideally this would be functionality which we integrate into the panel provided by the os patterns in some way so that the app doesn't need to handle it itself, but this seems to work reasonably for now (cc @thure)

Reply:
Yeah I noticed the sidebar would glitch when a new space was selected… I couldn't look into it at the time, but I'm glad you found a way to make it feel less broken while we flag it for investigation.